### PR TITLE
Added Loading Directive

### PR DESCRIPTION
**Pull request by patrickporto**
Added a simple directive for the loading behavior.

### CODE REVIEW

File: src/behaviors/loading/loading.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  .directive('smLoading', smLoading);
+
+  smLoading.$inject = ['$parse'];
+
+  function smLoading($parse) {
+    return {
+      restrict:'A',
+      link: function(scope, element, attrs) {
+        scope.$watch(function() {
```
Comment (caitp) [inline src/behaviors/loading/loading.js]: This is pretty great, I love how simple it is and how it doesn't allocate a new scope (despite allocating a new watch)

File: src/behaviors/loading/loading.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          return $parse(attrs.smLoading)(scope);
+        }, function(loading) {
+          if (loading) {
+            element.addClass('loading');
```
Comment (caitp) [inline src/behaviors/loading/loading.js]: Should the loading class be configurable?

File: src/behaviors/loading/demo/loading.module.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+(function() {
+    'use strict';
+
+    angular
+        .module('semantic.ui.angular.loading',[
+            'semantic.ui.behaviors.loading'
+        ]).run(function($rootScope) {
+
+          $rootScope.opts = {
```
Comment (caitp) [inline src/behaviors/loading/demo/loading.module.js]: this worries me a bit. We should probably not be polluting the root scope this way. How about registering a constant or variable in the module instead?

Comment (caitp) [inline src/behaviors/loading/demo/loading.module.js]: Wait, sorry my bad, thought this was part of the actual lirbary and not the demo. It's fine. But does anything actually use this?

File: src/behaviors/loading/loading.js (hunk 2 above)
Comment (patrickporto) [inline src/behaviors/loading/loading.js]: Have you a example?

Comment (caitp) [inline src/behaviors/loading/loading.js]: `<div sm-loading="ctrl.inProgress" class-name="loading-big"></div>` or something (or configurable defaults using the module.config() api)

File: src/behaviors/loading/demo/loading.module.js (hunk 3 above)
Comment (patrickporto) [inline src/behaviors/loading/demo/loading.module.js]: What? smLoading instead the ngClass?

File: src/behaviors/loading/loading.js (hunk 2 above)
Comment (patrickporto) [inline src/behaviors/loading/loading.js]: do would generalize behaviors, replacing the ngclass completely. Example using ngclass:
```html
<div ng-class="{'loading-big': ctrl.inProgress}"></div>
```

Comment (caitp) [inline src/behaviors/loading/loading.js]: Using this argument, you could say that there's no point in having an `smLoading` directive at all. I think what we want is a loading directive which is very simple, and very fast, not blocked on ngClass's animation sync stuff unless it's ever needed.

Comment (caitp) [inline src/behaviors/loading/loading.js]: Another API you could consider is actually using a promise directly to enable/disable the loading class, which is something ngClass definitely cannot do, and is more in tune with the way people are writing web apps. An angular2 style version using rx observables would be cool too

Comment (patrickporto) [inline src/behaviors/loading/loading.js]: In complex applications, this directive improvement the code readability, like a smButton directive. 
Example using ngClass:  
```html
<div class="ui segment" ng-class="{'loading': ctrl.cacheChecking}">
  <h1>Posts</h1>
  <div class="ui segment" ng-class="{'loading': ctrl.objects.length === 0}">
    <button class="ui button" ng-class="{'loading: ctrl.saving'}">
      Save
    </button>
  </div>
  
  <form class="ui form" ng-class="{'loading: ctrl.saving'}">
    <!-- ... -->
    <button class="ui button">
      Save
    </button>
  </form>
</div>
```

Example using smLoading:  
```html
<div class="ui segment" sm-loading="ctrl.cacheChecking">
  <h1>Posts</h1>
  <div class="ui segment" sm-loading="ctrl.objects.length === 0">
    <button class="ui button" sm-loading"ctrl.saving">
      Save
    </button>
  </div>
  
  <form class="ui form" sm-loading="ctrl.saving">
    <!-- ... -->
    <button class="ui button">
      Save
    </button>
  </form>
</div>
```

The loading class use is very common in application using semantic-ui and ajax, like Single Page Application.

Comment (caitp) [inline src/behaviors/loading/loading.js]: I know what it's useful for =) The point here is to try to get more value out of this directive. I think the promise API is a very good idea, specifically for the "ajax" use case you were discussing, specifically in angular. We can add it later, but it shouldn't be too difficult to add here

Comment (patrickporto) [inline src/behaviors/loading/loading.js]: having necessity, it's simple improve later. =)

**Pull request closed without merging**